reference-implementation/lib/readable-stream.js

```diff
@@ -484,3 +484,157 @@ function ShouldReadableStreamApplyBackpressure(stream) {
 
   return shouldApplyBackpressure;
 }
+
+function TeeReadableStream(stream, clone) {
+  const reader = stream.getReader();
+
+  let enqueue1, enqueue2, close1, close2, error1, error2;
+  let canceled1, cancelReason1, canceled2, cancelReason2;
+  const branch1 = new ReadableStream({
+    start(enqueue, close, error) {
+      [enqueue1, close1, error1] = [enqueue, close, error];
+    },
+    pull: readAndEnqueueInBoth,
+    cancel(reason) {
+      canceled1 = true;
+      cancelReason1 = reason;
+      maybeCancelSource();
+    }
+  });
+  const branch2 = new ReadableStream({
+    start(enqueue, close, error) {
+      [enqueue2, close2, error2] = [enqueue, close, error];
+    },
+    pull: readAndEnqueueInBoth,
+    cancel(reason) {
+      canceled2 = true;
+      cancelReason2 = reason;
+      maybeCancelSource();
+    }
+  });
+
+  return [branch1, branch2];
+
+  function readAndEnqueueInBoth() {
+    reader.read().then(
```

Comment:
Fixed. Watched it for errors only, otherwise the sequencing gets a bit screwy. I think that is correct since errors will clear the queue while closes will not.